dask/bag/core.py

```diff
@@ -7,19 +7,25 @@
 import os
 import uuid
 from warnings import warn
+from distutils.version import LooseVersion
 
 from ..utils import ignoring
 
 from toolz import (merge, take, reduce, valmap, map, partition_all, filter,
-        remove, compose, curry)
+                   remove, compose, curry, first, second, accumulate)
 from toolz.compatibility import iteritems, zip
 import toolz
+_implement_accumulate = LooseVersion(toolz.__version__) > '0.7.4'
```

Comment:
The trick is we need different versions for either cytoolz or pytoolz, and accumulate may come from either (or none) of them. It's a lot of work to get around the versioning shenanigans. Can nix all of this once dependencies are updated.